stores/json/store.go

```diff
@@ -19,43 +19,34 @@ type BinaryStore struct {
 	store Store
 }
 
-// Marshal takes a sops tree branch and returns a json formatted string
-func (store BinaryStore) Marshal(tree sops.TreeBranch) ([]byte, error) {
-	for _, item := range tree {
-		if item.Key == "data" {
-			return []byte(item.Value.(string)), nil
-		}
-	}
-	return nil, fmt.Errorf("No binary data found in tree")
+func (store BinaryStore) LoadEncryptedFile(in []byte) (sops.Tree, error) {
+	return store.store.LoadEncryptedFile(in)
 }
 
-// MarshalWithMetadata takes a sops tree branch and sops metadata and marshals them to json.
-func (store BinaryStore) MarshalWithMetadata(tree sops.TreeBranch, metadata sops.Metadata) ([]byte, error) {
-	return store.store.MarshalWithMetadata(tree, metadata)
+func (store BinaryStore) LoadPlainFile(in []byte) (sops.TreeBranch, error) {
```

Comment:
Basically the whole PR is name changes except this. In `master`, this would be done through `Unmarshal`, which would first try to load the file through the JSON store to check if it's encrypted (encrypted binary files are JSON). That's completely broken, because anything that seems like valid JSON will be loaded through the JSON store even though the binary store was desired.

With this PR, because the store knows whether the file being loaded is encrypted or not, there's no need to infer it through a hack like in `master`.